fall19/p10/test.py

```diff
@@ -2,19 +2,48 @@
 
 import json
 import os
-import subprocess
 import sys
-import importlib
-import inspect
-import traceback
 import re, ast, math
 from collections import namedtuple, OrderedDict, defaultdict
-from functools import wraps
 from bs4 import BeautifulSoup
 from datetime import datetime
 import nbconvert
 import nbformat
 
+try:
+    from lint import lint
+except ImportError:
+    err_msg = """Please download lint.py and place it in this directory for 
+    the tests to run correctly. If you haven't yet looked at the linting module, 
+    it is designed to help you improve your code so take a look at: 
+    https://github.com/tylerharter/cs301-projects/tree/master/linter"""
+    raise FileNotFoundError(err_msg)
+
+ALLOWED_LINT_ERRS = {
```

Comment:
maybe better to have dissalowed errors?

Do we care about E's?

What's the different between broad-except and bare-except?  When students don't specify an exception type, that seems like something we want to catch.

Reply:
There's upwards of 200 messages that can occur, so it seems reasonable to allow some and block the more obscure ones by default in order to be more conservative and not let messages that are ambiguous or disputable propagate out. But we certainly could have both a white and blacklist. I based this list on all the messages that occurred on the student's submissions from p3-p7. I don't think we'll get many surprises but I'll run the full linter after the fact to make sure.

broad-except is when students don't specify an exception type, bare-except is when there's no try that accompanies it.

As far as caring about E's, I think they are important. For instance, used-before-assignment, function-redefined, undefined-variable are all Es.